### PR TITLE
Bug Fix Public room: the avatar shape is wrong

### PR DESCRIPTION
**Pull request by giomfo**
Rewrite PublicRoomCell (Remove PublicRoomTableViewCell)
Rename MatrixIDComponents with UserIDComponents
Add RoomIDComponents

screenshot 2019-02-28 at 09 24 58
becomes
screenshot 2019-02-28 at 09 22 28

#152

### CODE REVIEW

File: Tchap/Modules/PublicRooms/Views/PublicRoomsCell.swift [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        
+        // Set the avatar
+        let avatarImage = AvatarGenerator.generateAvatar(forMatrixItem: publicRoom.roomId, withDisplayName: roomDisplayName)
+        
+        if let avatarUrl = publicRoom.avatarUrl {
+            self.avatarView.setImageURI(avatarUrl, withType: nil, andImageOrientation: UIImageOrientation.up, toFitViewSize: self.avatarView.frame.size, with: MXThumbnailingMethodCrop, previewImage: avatarImage, mediaManager: session.mediaManager)
+        } else {
+            self.avatarView.image = avatarImage
+        }
+        self.avatarView.contentMode = UIViewContentMode.scaleAspectFill
```
Comment (SBiOSoftWhare) [inline Tchap/Modules/PublicRooms/Views/PublicRoomsCell.swift]: You could shorter this expression by using `self.avatarView.contentMode = .scaleAspectFill`

File: Tchap/Modules/PublicRooms/Views/PublicRoomsCell.swift [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import UIKit
+
+@objcMembers class PublicRoomsCell: UITableViewCell, Stylable {
```
Comment (SBiOSoftWhare) [inline Tchap/Modules/PublicRooms/Views/PublicRoomsCell.swift]: You could conform to `NibReusable` from Reusable pod and remove those lines:
```
    static func nib() -> UINib {
        return UINib(nibName: String(describing: self), bundle: nil)
    }

    static func defaultReuseIdentifier() -> String {
        return String(describing: self)
    }
```

File: Tchap/Modules/PublicRooms/PublicRoomsDataSource.swift [hunk 3]
```diff
@@ -116,7 +116,7 @@ final class PublicRoomsDataSource: NSObject {
     // MARK: - Private
     
     private func setupTableViewCells() {
-        self.tableView?.register(PublicRoomTableViewCell.nib(), forCellReuseIdentifier: PublicRoomTableViewCell.defaultReuseIdentifier())
+        self.tableView?.register(PublicRoomsCell.nib(), forCellReuseIdentifier: PublicRoomsCell.defaultReuseIdentifier())
```
Comment (SBiOSoftWhare) [inline Tchap/Modules/PublicRooms/PublicRoomsDataSource.swift]: You could use `self.tableView?.register(PublicRoomTableViewCell.self)` if PublicRoomTableViewCell conform to NibReusable

File: Tchap/Modules/PublicRooms/PublicRoomsDataSource.swift [hunk 4]
```diff
@@ -146,11 +146,11 @@ extension PublicRoomsDataSource: UITableViewDataSource {
             
             let room = rooms[indexPath.row]
             
-            if let publicRoomCell = tableView.dequeueReusableCell(withIdentifier: PublicRoomTableViewCell.defaultReuseIdentifier(), for: indexPath) as? PublicRoomTableViewCell {
-                publicRoomCell.render(room, withMatrixSession: self.session)
+            if let publicRoomCell = tableView.dequeueReusableCell(withIdentifier: PublicRoomsCell.defaultReuseIdentifier(), for: indexPath) as? PublicRoomsCell {
```
Comment (SBiOSoftWhare) [inline Tchap/Modules/PublicRooms/PublicRoomsDataSource.swift]: you could use `let publicRoomCell = tableView .dequeueReusableCell(for indexPath: IndexPath, cellType: PublicRoomsCell.self)` if PublicRoomsCell conform to NibReusable

File: Tchap/Modules/PublicRooms/Views/PublicRoomsCell.swift [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        return String(describing: self)
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        self.avatarView.tc_makeHexagon(borderWidth: Constants.hexagonImageBorderWidth, borderColor: self.style.secondaryTextColor)
+    }
+    
+    func render(publicRoom: MXPublicRoom, withMatrixSession session: MXSession) {
```
[the remark this answers is not in the document]
Comment (giomfo) [reply Tchap/Modules/PublicRooms/Views/PublicRoomsCell.swift]: agree

File: Tchap/Modules/PublicRooms/Views/PublicRoomsCell.swift (hunk 1 above)
Comment (giomfo) [reply Tchap/Modules/PublicRooms/Views/PublicRoomsCell.swift]: ok

**Pull request merged**